go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -727,6 +767,27 @@ func (bsnc *BaseSecondaryNetworkController) WatchMultiNetworkPolicy() error {
 		return nil
 	}
 
+	if util.IsNetworkSegmentationSupportEnabled() && bsnc.IsPrimaryNetwork() {
```

Comment:
I guess througout the PR, why do we check `IsNetworkSegmentationSupportEnabled` along with `IsPrimaryNetwork`?

When don't do the equivalent with secondary networks checking `MultiNetworkEnabled` along with `IsSecondary()`

Reply:
Also, multinetpol should only apply to non primary networks?

Reply:
yes, i think multinetpol is applicable only for non primary networks.


> I guess througout the PR, why do we check `IsNetworkSegmentationSupportEnabled` along with `IsPrimaryNetwork`?
> 
This is a common practice being used all over the places. isn't this kind of gating check ? 
If this is not needed at secondary network controller level, can remove this.

Reply:
So if segmentation is not enabled, why woudl we have a network controller for a primary network? It shouldn't exist in the first place, already checked by the NAD controller. No harm I guess though.

Reply:
you are right, removed `IsNetworkSegmentationSupportEnabled` check whereever it's possible.